app/Http/Controllers/MasjidProfileController.php

```diff
@@ -39,4 +40,32 @@ public function update(Request $request): RedirectResponse
 
         return redirect()->route('masjid_profile.show');
     }
+
+    public function image(Request $request)
+    {
+        $this->authorize('edit_masjid_profile');
+
+        $validatedPayload = $request->validate([
+            'image' => 'required'
+        ]);
+
+        if (!base64_decode($validatedPayload['image'])) {
+            return response()->json([
+                'message' => 'Image not found'
+            ]);
+        }
+
+        $image_parts      = explode(";base64,", $request->image);
+        $image_base64     = base64_decode($image_parts[1]);
+        $image_name       = uniqid() . '.webp';
+        $image_full_path  = 'public/' . $image_name;
+
+        Storage::put($image_full_path, $image_base64);
+        Setting::set('masjid_image', $image_name);
+
+        return response()->json([
+            'message' => 'Image Uploaded Successfully',
+            'image' => asset(Setting::get('masjid_image'))
```

Comment:
Kita mesti pakai `Storage::url()` karena kita pakai `FILESYSTEM_DRIVER=public`

```suggestion
            'image' => Storage::url(Setting::get('masjid_image')),
```